packages/opencensus-exporter-stackdriver/src/stackdriver-stats-utils.ts

```diff
@@ -0,0 +1,288 @@
+      timeSeriesList.push({
+        metric: this.createMetric(
+            metricDescriptor, timeSeries.labelValues, metricPrefix),
+        resource: monitoredResource,
+        metricKind,
+        valueType,
+        points: timeSeries.points.map(point => {
+          return this.createPoint(point, timeSeries.startTimestamp, valueType);
+        })
+      });
+    }
+    return timeSeriesList;
+  }
+
+  /**
+   * Gets metric type.
+   * @param name The metric name.
+   * @param metricPrefix The metric prefix.
+   */
+  static getMetricType(name: string, metricPrefix: string): string {
+    return path.join(metricPrefix, name);
+  }
+
+  /**
+   * Creates Metric display name.
+   * @param name The metric name
+   * @param displayNamePrefix The metric display name.
+   */
+  static createDisplayName(name: string, displayNamePrefix: string): string {
+    return path.join(displayNamePrefix, name);
+  }
+
+  /**
+   * Converts a OpenCensus Type to a StackDriver MetricKind.
+   * @param metricDescriptorType The kind of metric.
+   */
+  static createMetricKind(metricDescriptorType: MetricDescriptorType):
+      MetricKind {
+    if (metricDescriptorType === MetricDescriptorType.GAUGE_INT64 ||
+        metricDescriptorType === MetricDescriptorType.GAUGE_DOUBLE) {
+      return MetricKind.GAUGE;
+    } else if (
+        metricDescriptorType === MetricDescriptorType.CUMULATIVE_INT64 ||
+        metricDescriptorType === MetricDescriptorType.CUMULATIVE_DOUBLE ||
+        metricDescriptorType === MetricDescriptorType.CUMULATIVE_DISTRIBUTION) {
+      return MetricKind.CUMULATIVE;
+    }
+    return MetricKind.UNSPECIFIED;
+  }
+
+  /**
+   * Converts a OpenCensus Type to a StackDriver ValueType.
+   * @param metricDescriptorType The kind of metric.
+   */
+  static createValueType(metricDescriptorType: MetricDescriptorType):
+      ValueType {
+    if (metricDescriptorType === MetricDescriptorType.CUMULATIVE_DOUBLE ||
+        metricDescriptorType === MetricDescriptorType.GAUGE_DOUBLE) {
+      return ValueType.DOUBLE;
+    } else if (
+        metricDescriptorType === MetricDescriptorType.GAUGE_INT64 ||
+        metricDescriptorType === MetricDescriptorType.CUMULATIVE_INT64) {
+      return ValueType.INT64;
+    } else if (
+        metricDescriptorType === MetricDescriptorType.GAUGE_DISTRIBUTION ||
+        metricDescriptorType === MetricDescriptorType.CUMULATIVE_DISTRIBUTION) {
+      return ValueType.DISTRIBUTION;
+    } else {
+      return ValueType.VALUE_TYPE_UNSPECIFIED;
+    }
+  }
+
+  /**
+   * Constructs a LabelDescriptor from a LabelKey.
+   * @param tag The Tags to get TimeSeries information from.
+   */
+  static createLabelDescriptor(labelKeys: LabelKey[]): LabelDescriptor[] {
+    const labelDescriptorList = labelKeys.map(labelKey => {
+      return {
+        key: labelKey.key,
+        valueType: 'STRING',  // Now we only support String type.
+        description: labelKey.description
+      } as LabelDescriptor;
```

Comment:
Could you use a type annotation on `labelDescriptorList`, i.e. `const labelDesceriptorList: LabelDescriptor[] = ` rather than a type assertion `as LabelDescriptor` here? The type annotation will cause the code to fail to compile if the interface changes so would help detect refactoring bugs.